src/node.cc

```diff
@@ -1151,6 +1151,8 @@ Local<Value> MakeCallback(Environment* env,
 
   Local<Function> pre_fn = env->async_hooks_pre_function();
   Local<Function> post_fn = env->async_hooks_post_function();
+  Local<Object> process = env->process_object();
```

Comment:
Perhaps move this down to where it's first needed?  It creates a new entry in the current HandleScope, which has a small but non-zero cost.